lib/internal/streams/end-of-stream.js

```diff
@@ -9,8 +9,13 @@ const {
 } = require('internal/errors').codes;
 const { once } = require('internal/util');
 
-function isRequest(stream) {
-  return stream.setHeader && typeof stream.abort === 'function';
+function isLegacyRequest(stream) {
+  return stream.setHeader && typeof stream.abort === 'function' &&
+    typeof stream.writableFinished !== 'boolean';
+}
+
+function isLegacyStream(stream) {
+  return typeof stream.writableFinished !== 'boolean';
```

Comment:
this is not the same as before, I'd keep the check on `_writableStream` for backward compatibility when shipping this in readable-stream?

Reply:
Ah, I did not consider readable-stream